tools/hsmtools.c

```diff
@@ -0,0 +1,135 @@
+#include <bitcoin/privkey.h>
+#include <ccan/str/str.h>
+#include <ccan/err/err.h>
+#include <ccan/noerr/noerr.h>
+#include <ccan/read_write_all/read_write_all.h>
+#include <common/utils.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <sodium.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define ERROR_HSM_FILE errno
+#define ERROR_USAGE 2
+#define ERROR_LIBSODIUM 3
+
+static void show_usage(void)
+{
+	printf("./hsmtools <method> [arguments]\n");
+	printf("methods:\n");
+	printf("	- decrypthsm <path/to/hsm_secret> <password>\n");
+	exit(0);
+}
+
+static bool ensure_hsm_secret_exists(int fd)
+{
+	if (fsync(fd) != 0) {
+		close(fd);
+		return false;
+	}
+	if (close(fd) != 0)
+		return false;
+
+	fd = open(".", O_RDONLY);
```

Comment:
This is fsyncing the wrong dir?  You want dirname(hsm_secret_path) here.

Reply:
Bad copy/paste!